lib/src/document.dart

```diff
@@ -14,6 +14,10 @@ import 'util.dart';
 /// Maintains the context needed to parse a Markdown document.
 class Document {
   final Map<String, LinkReference> linkReferences = {};
+  // keys are case sensitive.
```

Comment:
Nit: Start with a capital letter [1] and hyphenate 'case-sensitive'.

[1] https://dart.dev/guides/language/effective-dart/documentation#do-format-comments-like-sentences